benchmark/httpapi.py

```diff
@@ -114,10 +138,12 @@ def post(self, request):
         request.setHeader(b'content-type', b'application/json')
         try:
             json = loads(request.content.read())
+            json['userdata']['branch']
```

Comment:
After discussion with @jongiddy, we don't think it should be required for the user to specify a branch when submitting a new set of benchmarking results. If the branch is available for a particular set of data then it can be included in any query that includes a branch, otherwise it can be ignored or retrieved by its ID.

This came up because of the error message that follows. If `branch` is not included then the message will state that `branch` is missing but doesn't make it clear to the user that it is actually missing from `userdata`. We may not want to validate the `userdata` section.